src/Compilers/Core/Portable/InternalUtilities/UnicodeCharacterUtilities.cs

```diff
@@ -109,6 +109,27 @@ public static bool IsValidIdentifier(string name)
             return true;
         }
 
+        /// <summary>
+        /// Returns true if every identifier in a namespace is valid using <see cref="IsValidIdentifier(string)"/>
+        /// </summary>
+        public static bool IsValidNamespace(string @namespace)
```

Comment:
If you're adding this at the compiler layer, please add some tests for it as well. Ping @gafter to review this as well.

Reply:
We don't seem to have direct tests for many of the utilities. I'm happy to add if it makes sense, but only for this specific method. Others will still be untested afaict.

I need to add some tests for the viewmodal as well, which would exercise this method. Whatever works best for the compiler team.